infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/environment/OpenShiftEnvironmentFactory.java

```diff
@@ -146,6 +159,32 @@ protected OpenShiftEnvironment doCreate(
     return osEnv;
   }
 
+  @VisibleForTesting
+  void setRamLimitAttribute(Map<String, InternalMachineConfig> machines, Collection<Pod> pods) {
+    for (Pod pod : pods) {
+      for (Container container : pod.getSpec().getContainers()) {
+        final String machineName = Names.machineName(pod, container);
+        final InternalMachineConfig machineConfig = machines.get(machineName);
```

Comment:
I think it is possible that machineConfig is null. Wouldn't we get NPE in that case?